bindings/c/test/mako/mako.c

```diff
@@ -801,28 +908,73 @@ void* worker_thread(void* thread_args) {
 
 	/* build/popualte */
 	else if (args->mode == MODE_BUILD) {
-		rc = populate(transaction, args, worker_id, thread_id, thread_tps, stats);
+		rc =
+		    populate(transaction, args, worker_id, thread_id, thread_tps, stats, block, elem_size, is_memory_allocated);
 		if (rc < 0) {
 			fprintf(stderr, "ERROR: populate failed\n");
 		}
 	}
 
 	/* run the workload */
 	else if (args->mode == MODE_RUN) {
-		rc = run_workload(transaction, args, thread_tps, throttle_factor, thread_iters, signal, stats, dotrace);
+		rc = run_workload(transaction, args, thread_tps, throttle_factor, thread_iters, signal, stats, dotrace, block,
+		                  elem_size, is_memory_allocated);
 		if (rc < 0) {
 			fprintf(stderr, "ERROR: run_workload failed\n");
 		}
 	}
 
+	if (args->mode == MODE_BUILD || args->mode == MODE_RUN) {
+		char str2[1000];
+		sprintf(str2, "/dev/shm/tempDataFile%d", *parent_id);
+		rc = mkdir(str2, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+		for (op = 0; op < MAX_OP; op++) {
+			if (args->txnspec.ops[op][OP_COUNT] > 0 || op == OP_COMMIT || op == OP_TRANSACTION) {
+				FILE* fp;
+				char file_name[NAME_MAX] = { '\0' };
+				strcat(file_name, str2);
+				get_file_name(file_name, worker_id, thread_id, op);
+				fp = fopen(file_name, "w");
+				lat_block_t* temp_block = ((thread_args_t*)thread_args)->block[op];
+				if (is_memory_allocated[op]) {
```

Comment:
I'm not sure how this is working.  Are we looking at the very last block only?

Reply:
We are looking at the very first block.